iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/OperationalLimitsGroupImpl.java

```diff
@@ -101,26 +110,34 @@ public void setApparentPowerLimits(ApparentPowerLimits limits) {
     }
 
     public Validable getValidable() {
-        return identifiable;
+        return validable;
     }
 
     public void notifyPermanentLimitUpdate(LimitType limitType, double oldValue, double newValue) {
         PermanentLimitInfo oldPermanentLimitInfo = new PermanentLimitInfo(oldValue, id, id.equals(selectedGroupId));
         PermanentLimitInfo newPermanentLimitInfo = new PermanentLimitInfo(newValue, id, id.equals(selectedGroupId));
-        identifiable.getNetwork().getListeners().notifyUpdate(identifiable, attributeName + "_" + limitType + ".permanentLimit",
-                oldPermanentLimitInfo, newPermanentLimitInfo);
+        doNotify(attributeName + "_" + limitType + ".permanentLimit", oldPermanentLimitInfo, newPermanentLimitInfo);
     }
 
     private void notifyUpdate(LimitType limitType, OperationalLimits oldValue, OperationalLimits newValue) {
         OperationalLimitsInfo oldOperationalLimitsInfo = new OperationalLimitsInfo(oldValue, id, id.equals(selectedGroupId));
         OperationalLimitsInfo newOperationalLimitsInfo = new OperationalLimitsInfo(newValue, id, id.equals(selectedGroupId));
-        identifiable.getNetwork().getListeners().notifyUpdate(identifiable, attributeName + "_" + limitType,
-                oldOperationalLimitsInfo, newOperationalLimitsInfo);
+        doNotify(attributeName + "_" + limitType, oldOperationalLimitsInfo, newOperationalLimitsInfo);
+    }
+
+    protected boolean notificationEnabled() {
+        return true;
+    }
+
+    private void doNotify(String attribute, Object oldValue, Object newValue) {
+        if (notificationEnabled() && identifiable instanceof AbstractIdentifiable<?> ident) {
+            ident.getNetwork().getListeners().notifyUpdate(identifiable, attribute, oldValue, newValue);
+        }
```

Comment:
I'm not very fond of the condition. If we really ask in the constructor a parameter for each functionality, we could add a NetworkListenerList as parameter.

Reply:
And then if we do that, we only need the id for the identifiable, hence the three fields: String ownerId, NetworkListenerList listeners, Validable validable

Reply:
Identifiable is still needed for the notify

Reply:
You are right, this is way cleaner like this. I just pushed a change.